### ISSUE TITLE
Array creation is intrinsically non-null

### ISSUE THREAD

**Issue by wmdietl** [eisop/checker-framework#927]
I noticed when working on https://github.com/jspecify/jspecify/pull/678 that the Nullness Checker currently accepts `@Nullable` on the created array:

````
import org.checkerframework.checker.nullness.qual.Nullable;

public class ADemo {
  void foo() {
    int [] o = new int @Nullable [10];
    o.hashCode();
  }
}
````

An error is raised on the `o.hashCode()` even though that reference is intrinsically non-null.

Like for object creation, an array creation cannot be `@Nullable`. (Check other possible qualifiers, also initialization).

**Comment by thisisalexandercook** [comment]
@wmdietl I'm unable to reproduce the error raised on `o.hashCode` but will work on updating checker to not accept qualifier on array creation.

**Comment by wmdietl** [comment]
Ah, I had used a 2 year old CF version... that produced:

````
> ./checker/bin/javac -processor nullness ADemo.java
ADemo.java:6: error: [dereference.of.nullable] dereference of possibly-null reference o
    o.hashCode();
    ^
1 error
````

It doesn't really make sense, because the logic that [makes array creations non-null](https://github.com/eisop/checker-framework/blob/79827b2877c753d32f36d8057f6dc833d3a92356/checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitAnnotatedTypeFactory.java#L745) has been there for a long while.

Let's just produce a warning that the `@Nullable` in the example is ignored.
